forecast_solar/__init__.py

```diff
@@ -109,7 +109,7 @@ async def _request(
         if response.status < 500:
             self.ratelimit = Ratelimit.from_response(response)
 
-        if response.status == 400:
+        if response.status in [400, 403]:
```

Comment:
We don't seem to handle any of these library specific exceptions in the integration. We may want to raise a specific exception if authentication fails, so the library user can ask the end user to re-authenticate.

Reply:
Something like this? 

```py
class ForecastSolarAuthenticationError(ForecastSolarError):
    """Forecast.Solar API authentication exception."""

    def __init__(self, data: dict) -> None:
        """Init a solar auth error.

        https://doc.forecast.solar/doku.php?id=api#invalid_request
        """
        super().__init__(f'{data["text"]} (error {data["code"]})')
        self.code = data["code"]
```

Reply:
Yes. We normally start a reauth flow by raising `ConfigEntryAuthFailed` in the integration. See existing integrations that do this.

Reply:
![image](https://user-images.githubusercontent.com/20448157/158197472-f962397d-92a1-425d-90c4-f43f2d10e0fc.png)

I think only if you actually went over the rate limit?

Reply:
Ok, so maybe move that code under that specific exception and remove the general 500 case.

Reply:
Maybe we should save this for another pull request and see what we want to do with the ratelimit in the end. Currently it is only used in the example.py at a status of 200 to see what your limit is. At status 429, which is also about the rate limit, you will see other data.

Reply:
Ok.